src/main/java/org/sqlite/SQLiteConnection.java

```diff
@@ -421,7 +456,9 @@ public void commit() throws SQLException {
         checkOpen();
         if (connectionConfig.isAutoCommit()) throw new SQLException("database in auto-commit mode");
         db.exec("commit;", getAutoCommit());
-        db.exec(connectionConfig.transactionPrefix(), getAutoCommit());
+        db.exec(this.transactionPrefix(), getAutoCommit());
+        this.firstStatementWasExecuted = false;
+        this.setCurrentTransactionType(this.getConnectionConfig().getTransactionMode());
```

Comment:
no need to use `this.getConnectionConfig()`, could use `connectionConfig` instead, as it is only set in the constructor and it is marked `final`